test/support/rekor_helper.rb

```diff
@@ -59,4 +68,101 @@ def build_rekord_entry(options)
       }
     }.deep_merge(options)
   end
+
+  def rekor_index_retrieve_url(uuid)
+    rekor_api_url('index', 'retrieve')
+  end
+
+  def stub_rekor_search_index_by_digest(gem_path: @gem_path, uuid: "dummy_entry_uuid", body: {}, returning: nil)
+    gem = Gem::Sigstore::Gemfile.new(gem_path)
+
+    stub_request(:post, rekor_index_retrieve_url(uuid))
+      .with(
+        headers: {
+          content_type: 'application/json',
+        },
+        body: {
+          hash: "sha256:#{gem.digest}",
+        },
+      )
+      .to_return_json(returning || ["dummy_entry_uuid"])
+  end
+
+  def rekor_log_entry_url(uuid)
+    rekor_api_url('log', 'entries', uuid)
+  end
+
+  def stub_rekor_get_rekord_by_uuid(
+    gem_path: @gem_path,
+    uuid: "dummy_entry_uuid",
+    log_entry_options: {},
+    rekord_options: {}
+  )
+    stub_request(:get, rekor_log_entry_url(uuid))
+      .to_return_json(
+        build_rekord_log_entry(
+          uuid: uuid,
+          log_entry_options: log_entry_options,
+          rekord_options: rekord_options,
+          gem_path: gem_path
+        )
+      )
+  end
+
+  def build_rekord_log_entry(uuid:, log_entry_options:, rekord_options:, gem_path:)
+    {
+      uuid => {
+        body: Base64.encode64(build_rekord(rekord_options, gem_path).to_json),
+        integratedTime: 1637154947,
+        logID: "dummy rekord logID",
+        logIndex: 864991,
+        verification: {
+          signedEntryTimestamp: "dummy timestamp signature"
+        }
+      }
+    }.deep_merge(log_entry_options)
+  end
+
+  def build_rekord(rekord_options, gem_path)
+    gem = Gem::Sigstore::Gemfile.new(gem_path)
+    pkey = Gem::Sigstore::PKey.new
+    cert_chain = build_fulcio_cert_chain(pkey.public_key)
+    stub_get_ca_certificate(certificate: cert_chain.first)
```

Comment:
Stubbing here kinda sucks, as it is unexpected.  The root cert built in `build_fulcio_cert_chain` currently varies in two ways:
- the CA signing key pair is generated on every call;
- the not-before and not-after values relate to `Time.now`.
We could hard-code a key pair in our fixures to solve the first problem. We could solve # 2 by providing very long-lived default time boundaries.  I'm ok with deferring such an improvement to later.

In the meantime, tests are free to re-stub `stub_get_ca_certificate` with a different response code/body in tests that require it.

Reply:
> by providing very long-lived default time boundaries. 

Have you thought of freezing time with something like [timecop](https://github.com/travisjeffery/timecop)?